src/models/team-application-access.ts

```diff
@@ -0,0 +1,239 @@
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 */
+	const get = async function (
+		teamApplicationAccessId: number,
+		options: BalenaSdk.PineOptions<BalenaSdk.TeamApplicationAccess> = {},
+	): Promise<BalenaSdk.TeamApplicationAccess | undefined> {
+		const teamApplicationAccess = await sdkInstance.pine.get({
+			resource: 'team_application_access',
+			id: teamApplicationAccessId,
+			options,
+		});
+		if (teamApplicationAccess == null) {
+			throw new Error('Team application access not found');
+		}
+		return teamApplicationAccess;
+	};
+
+	/**
+	 * @summary Add applications access to team
+	 * @name add
+	 * @public
+	 * @function
+	 * @memberof balena.models.team.applicationAccess.add
+	 *
+	 * @description This method add application access to team.
+	 *
+	 * @param {Number} teamId - Required: the team id the application access will be granted for.
+	 * @param {Number|String} applicationIdOrSlug - Required: application id or slug
+	 * @param {String} RoleName - Required: application membership role name (ApplicationMembershipRoles)
+	 *
+	 * @fulfil {Object} - TeamApplicationAccess
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.applicationAccess.add(1239948, 'MyAppSlug', 'developer').then(function(teamApplicationAccess) {
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 *
+	 * @example
+	 * balena.models.team.applicationAccess.add(1239948, 456789, 'observer').then(function(teamApplicationAccess) {
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 */
+	const add = async function (
+		teamId: number,
+		applicationIdOrSlug: number | string,
+		roleName: BalenaSdk.ApplicationMembershipRoles,
+	): Promise<BalenaSdk.TeamApplicationAccess> {
+		const appId = (
+			await sdkInstance.models.application.get(applicationIdOrSlug, {
+				$select: 'id',
+			})
+		)?.id;
+
+		if (appId == null) {
+			throw new errors.BalenaApplicationNotFound(applicationIdOrSlug);
+		}
+		const roleId = await getRoleId(roleName);
+
+		return sdkInstance.pine.post({
+			resource: 'team_application_access',
+			body: {
+				team: teamId,
+				grants_access_to__application: appId,
+				application_membership_role: roleId,
+			},
+		});
+	};
+
+	/**
+	 * @summary Update team application access
+	 * @name update
+	 * @public
+	 * @function
+	 * @memberof balena.models.team.applicationAccess
+	 *
+	 * @description This method update a team application access role.
+	 *
+	 * @param {Number} teamApplicationAccessId - Required: the team application access id.
+	 * @param {String} roleName - Required: The new role to assing (ApplicationMembershipRoles).
+	 *
+	 * @fulfil {Object} - TeamApplicationAccess
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.update(123, 'developer').then(function(teamApplicationAccess) {
+	 * 	console.log(teamApplicationAccess);
+	 * });
+	 */
+	const update = async function (
+		teamApplicationAccessId: string | number,
```

Comment:
Regarding the internal server error: the same issue applies here for `teamApplicationAccessId`